kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/identity/FileStoredIdentityManager.java

```diff
@@ -0,0 +1,94 @@
+package org.kie.server.router.identity;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import io.undertow.security.idm.Account;
+import io.undertow.security.idm.Credential;
+import io.undertow.security.idm.PasswordCredential;
+import org.jboss.logging.Logger;
+
+public class FileStoredIdentityManager implements IdentityService {
+
+    protected static final Logger log = Logger.getLogger(FileStoredIdentityManager.class);
+
+    private File userFile;
+
+    public FileStoredIdentityManager() {
+        userFile = new File("users.properties");
+        log.infof("Using <%1$s> as user repository", userFile.getAbsolutePath());
+
+    }
+
+    @Override
+    public Account verify(Account account) {
+        return account;
+    }
+  
+    @Override
+    public Account verify(Credential credential) {
+        return null;
+    }
+  
+    @Override
+    public Account verify(String id, Credential credential) {
+        if (!(credential instanceof PasswordCredential)) {
+            return null;
+        }
+        PasswordCredential passwordCredential = (PasswordCredential) credential;
+        KieServerInstanceAccount account = getAccount(id);
+        String challenge = HashUtil.hash(id, new String(passwordCredential.getPassword()));
+        if (account != null && challenge.equals(account.getHash())) {
+            log.infof("Authentication succesfully for %1$2", id);
+            return account;
+        }
+        log.infof("Authentication failed for %1$2", id);
+        return null;
+    }
+
+    private KieServerInstanceAccount getAccount(String id) {
+        Map<String, KieServerInstanceAccount> users = new HashMap<>();
+        Properties file = loadKieServerInstanceData();
+        file.stringPropertyNames().forEach(name -> users.put(name, new KieServerInstanceAccount(name, file.getProperty(name))));
+        return users.get(id);
+    }
+
+    @Override
+    public void addKieServerInstance(String kieServerInstanceId, String password) throws Exception {
+        Properties data = loadKieServerInstanceData();
+        data.put(kieServerInstanceId, HashUtil.hash(kieServerInstanceId, password));
+        saveKieServerInstanceData(data);
+    }
+
+    @Override
+    public void removeKieServerInstance(String kieServerInstanceId) {
+        Properties data = loadKieServerInstanceData();
+        data.remove(kieServerInstanceId);
+        saveKieServerInstanceData(data);
+    }
+
+    private Properties loadKieServerInstanceData() {
+        Properties properties = new Properties();
+        try (InputStream is = new FileInputStream(userFile)) {
+            properties.load(is);
+        } catch (Exception e) {
+            log.errorf("file not found %1$s", userFile);
```

Comment:
Please separate the FileNotFoundException exception handling from other IO exceptions.
FileNotFoundException can be expected in case user creates first entry, in this case warning could be enough, saying that file was not found, empty instance data are returned.
For other IO exception I would leave there error severity.